casepro/msgs/tests.py

```diff
@@ -1211,6 +1211,78 @@ def test_search(self):
             'user_assignee': {'id': self.user1.pk, 'name': "Evan"}})
         self.assertEqual(response.json['results'][1]['id'], 104)
 
+        t1 = now()
+        self.create_message(self.unicef, 210, self.ann, "Is this thing on?", [self.aids], is_handled=True)
+        Message.bulk_flag(self.unicef, self.user1, [msg5])
+
+        t2 = now()
+
+        # test the refresh, only new items and changes after the date is returned
+        response = self.url_get('unicef', url, {
+            'folder': 'inbox', 'text': "", 'page': 1, 'after': format_iso8601(t1), 'last_refresh': format_iso8601(t1),
+            'before': format_iso8601(t2)
+        })
+
+        self.assertEqual(len(response.json['results']), 2)
+        self.assertEqual(response.json['results'][0]['id'], 210)
+        self.assertEqual(response.json['results'][1]['id'], 105)
+        self.assertEqual(response.json['results'][1]['flagged'], True)
+
+    def test_get_lock(self):
+        msg = self.create_message(self.unicef, 101, self.ann, "Normal", [self.aids, self.pregnancy])
+
+        # The message is not locked
+        with self.assertNumQueries(0):
```

Comment:
This isn't really checking anything since self.user1 is obviously fetched. What I meant was checking that when we fetch multiple messages over the search endpoint, we don't have to db hits for each message because we haven't prefetched locked_by. You can remove these and I'll get this merged.